draft-gpew-priv-ppm.md

```diff
@@ -628,268 +653,642 @@ encoded value of the following form:
 
 Once a set of clients have uploaded their reports to the leader, the leader can
 send them to the helpers to be verified and aggregated. In order to enable the
 system to handle very large batches of reports, this process can be performed
-incrementally. To aggregate a set of reports, the leader sends an AggregateReq
-to each helper containing those report shares. The helper then processes them
-(verifying the proofs and incorporating their values into the ongoing aggregate)
-and replies to the leader.
+with multiple set of reports in parallel. To aggregate a set of reports, the
+leader sends a requet to each helper containing those report shares. Each helper
+then processes them (verifying the proofs and incorporating their values into
+the ongoing aggregate) and replies to the leader.
 
-The exact structure of the aggregation flow depends on the VDAF.  Specifically:
+The exact structure of the aggregation flow depends on the VDAF. Specifically:
 
 * Some VDAFs (e.g., `prio3`) allow the leader to start aggregating reports
   proactively before all the reports in a batch are received. Others (e.g.,
   `poplar1`) require all the reports to be present and must be initiated by the
-  collector.
+  collector. (See {{collect-flow}}.)
 
-* Processing the reports -- especially validating them -- may require multiple
-  round trips.
+* Processing a set of reports -- especially validating them -- may require
+  multiple round trips.
 
 Note that it is possible to aggregate reports from one batch while reports from
 the next batch are coming in. This is because each report is validated
 independently.
 
-This process is illustrated below in {{pa-aggregate-flow}}. In this example, the
-batch size is 20, but the leader opts to process the reports in sub-batches of
-10. Each sub-batch takes two round-trips to process. Once both sub-batches have
-been processed, the leader can issue an AggregateShareReq in order to retrieve
-the helper's aggregated result.
+This process is illustrated below in {{aggregation-flow-illustration}}. In this
+example, the batch size is 20, but the leader opts to process the reports in
+sub-batches of 10. Each sub-batch takes two round-trips to process. Once both
+sub-batches have been processed, the leader can issue an AggregateShareReq in
+order to retrieve the helper's aggregate share. (see
+{{aggregate-share-request}}.)
 
 In order to allow the helpers to retain minimal state, the helper can attach a
 state parameter to its response, with the leader returning the state value in
-the next request, thus offloading the state to the leader. This state value MUST
-be cryptographically protected as described in {{helper-state}}.
+the next request, thus offloading the state to the leader. The state is not
+carried across runs of the aggregation protocol, i.e., the initial aggregate
+request sent by the leader in a given run does not contain helper state.
+[CP: This diverges from the original intent of the helper state, which was to
```

Comment:
x-posting from another comment about helper state:

I said elsewhere I think we should remove the helper state blob unless we can prove it's necessary. However if we keep it, we should put in a SHOULD or RECOMMENDED about versioning inside the state blob, because what if the helper deploys a new version while the leader is holding onto a state blob?

In passing, absolving helpers of dealing with state blob versioning and format migration is another strong argument for removing that feature from the protocol.